osf/migrations/0109_set_ever_public.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.13 on 2018-06-08 15:01
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+from osf.utils.workflows import DefaultStates
+
+def set_ever_public(state, *args, **kwargs):
+    PreprintService = state.get_model('osf', 'preprintservice')
+    for preprint in (PreprintService.objects.filter(is_published=True).
+            exclude(models.Q(provider__reviews_workflow='post-moderation') &
+                models.Q(machine_state=DefaultStates.PENDING.value))):
+
+        preprint.ever_public = True
+        preprint.save()
```

Comment:
This migration could be optimized by replacing the `for` loop with something like
```
PreprintService.objects.filter(<target QS>).update(ever_public=True)
```
This causes the DjangORM to make a single, larger [`UPDATE`](https://www.postgresql.org/docs/10/static/sql-update.html) query, rather than one per updated preprint.

Reply:
I read the bulk update don't call custom `save`, but I guess since migration does not need to benefit from future updates to custom `save` it should be fine.